### PR TITLE
firmware update

### PR DESCRIPTION
**Pull request by Srujana-2000**
##### SUMMARY

New module_util request
Adding a new module_utils file for the purpose of Updating, Getting Info, fetching the Power_Status.
And the addition of three module files separately for Updating, fetching Inventory Info and Getting or modifying the power status.

##### ISSUE TYPE

- New Module/Plugin Pull Request

##### COMPONENT NAME

module_utils\hpc_system_firmware_utils.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Srujana-2000** [comment]
Can we know what is the further process for the following merge?

**Comment by Srujana-2000** [comment]
Just adding a remainder for help of the following merge

**Comment by felixfontein** [comment]
@Srujana-2000 I suggest you take a look at https://github.com/ansible-collections/community.general/blob/main/CONTRIBUTING.md#creating-new-modules-or-plugins (again). New modules must come with tests. This PR does not contain any tests. Also adding more than one module in a PR is usually not a good idea. Finally, as I already wrote, since this is a contribution of a company to a community-driven collection, and does not rely on any existing content in this collection, it's better to create a new collection for this instead of adding it to this collection.

**Comment by Srujana-2000** [comment]
As per the comments I have added tests. Can we get confirmed if it's correct. And move forward for further process

**Comment by Srujana-2000** [comment]
We want to be in community general rather than creating a separate collection. Can we pls get confirmed about this

**Comment by felixfontein** [comment]
Can you explain *why* you would prefer the modules to be here instead of a separate collection?

**Comment by Srujana-2000** [comment]
Our products are already being supported through the ilo_redfish utils and we want to extent the support for firmware upgrade by being in line with them. Also community general provides redfish support, which is the core part of the upgrade so we wanted to be in community general collection

**Comment by Srujana-2000** [comment]
Our products are already being supported through library inside community general and we want to extend the support for firmware upgrade by being in line with them. Also community general provides redfish support and other dependency, which is the core part of the upgrade so we wanted to be in community general collection. We would not prefer to again rewrite all the code for separate collection

**Comment by Srujana-2000** [comment]
Waiting for your reply

**Review by russoz — COMMENTED** [review]
Given this modules/utils seem to rely on redfish APIs, I would argue that they should be added to `plugins/module_utils/redfish_utils.py` or `plugins/module_utils/ilo_redfish_utils.py`.

The way this is going, every vendor will have their own redfish-related files, and I am sure that there would be a lot of overlap amongst them.

File: .github/BOTMETA.yml [hunk 1]
```diff
@@ -610,6 +610,12 @@ files:
     maintainers: jameslivulpi
   $modules/honeybadger_deployment.py:
     maintainers: stympy
+  $modules/hpc_get_power_state.py:
```
[the remark this answers is not in the document]
Comment (Srujana-2000) [reply .github/BOTMETA.yml:613]: Added maintainer for module_utils

[the remark this answers is not in the document]
Comment (Srujana-2000) [reply .github/BOTMETA.yml:613]: Thank you for your suggestion. While I appreciate the concern regarding potential overlap among Redfish-related files, I believe there are valid reasons for keeping separate module utility files for different vendors' Redfish implementations.

Firstly, the functions and operations required for managing servers through Redfish APIs can vary significantly between different vendors and even different product lines within the same vendor. Consolidating all Redfish-related utilities into a single file could lead to confusion and unnecessary complexity, especially when dealing with distinct sets of functions tailored to specific vendor requirements.

Moreover, maintaining separate files allows for clearer organization and easier maintenance. It enables us to focus on developing and optimizing functions specific to each vendor's Redfish implementation without being bogged down by irrelevant or conflicting code from other vendors.

In summary, while I understand the concern about potential redundancy, I believe that maintaining separate module utility files for different vendors' Redfish implementations ultimately promotes clarity, maintainability, and flexibility in our codebase.

Let me know if you have any further questions or concerns regarding this approach.
